test/unit/test_loader.py

```diff
@@ -144,9 +144,12 @@ def test_datastore_load_successful(self, post):
         mock_post.json.side_effect = [
             {'success': True, 'result': {'id': 1}},
             {'success': True, 'result': {'resource_id': 1}},
+            {'success': True, 'result': {'id': 1}},
```

Comment:
:+1: 

I would recommend breaking this out into its own test, though, and maybe adding some additional test cases for both about error handling, etc.